### PR TITLE
Merge 1.2

### PR DESCRIPTION
**Pull request by dlvenable**
### Description

Merges the `1.2` branch (from commit `e3d5b673`) into `main`.
 
### Issues Resolved

N/A
 
### Check List
- [x] New functionality includes testing.
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by graytaylor0 — APPROVED** [review]

**Review by sshivanii — DISMISSED** [review]
DCO check is failing for some commits where emails were changed but we can ignore them.

File: data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/source/RandomStringSourceTests.java [hunk 1]
```diff
@@ -22,6 +22,9 @@
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.greaterThan;
 
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.greaterThan;
```
Comment (dinujoh) [inline data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/source/RandomStringSourceTests.java:26]: duplicate import. I see them in line 22-23.

File: data-prepper-plugins/http-source/src/main/java/com/amazon/dataprepper/plugins/source/loghttp/HTTPSource.java [hunk 2]
```diff
@@ -80,6 +80,9 @@ public void start(final Buffer<Record<Log>> buffer) {
         }
         if (server == null) {
             final ServerBuilder sb = Server.builder();
+
+            sb.disableServerHeader();
```
Comment (dinujoh) [inline data-prepper-plugins/http-source/src/main/java/com/amazon/dataprepper/plugins/source/loghttp/HTTPSource.java:84]: Is this related to log4j CVE ?

Comment (dlvenable) [reply data-prepper-plugins/http-source/src/main/java/com/amazon/dataprepper/plugins/source/loghttp/HTTPSource.java:84]: No, this is not related to log4j. It was added as an additional security protection to prevent unauthorized persons from know what type of server is in use.

**Review by sshivanii — APPROVED** [review]

**Review by dinujoh — APPROVED** [review]
